utils/watermark_disk.go

```diff
@@ -0,0 +1,114 @@
+package utils
+
+import (
+	"crypto/rand"
+	"fmt"
+	"io"
+	"math/big"
+	"os"
+	"slices"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	bufferSize    = 512
+	numWatermarks = 10
+)
+
+type watermark struct {
+	position int64
+	data     []byte
+}
+
+func ApplyWatermarks(logicalName string) func() error {
```

Comment:
```suggestion
func ApplyWatermarks(logicalName string) (func() error, error) {
```
Sorry if I misled you before but I meant that cleanup func should return error not bool, but we should still return errors in this function too for the errors that are hit in this func